framework/auth/views.py

```diff
@@ -36,95 +36,72 @@
 
 
 @collect_auth
-def reset_password_get(auth, verification_key=None, **kwargs):
+def reset_password_get(auth, uid=None, token=None):
     """
     View for user to land on the reset password page.
     HTTp Method: GET
 
-    :raises: HTTPError(http.BAD_REQUEST) if verification_key is invalid
+    :param auth: the authentication state
+    :param uid: the user id
+    :param token: the token in verification key
+    :return
+    :raises: HTTPError(http.BAD_REQUEST) if verification key for the user is invalid or has expired
     """
 
-    # If user is already logged in, log user out
+    # TODO: discuss this with @MattF
+    # if users are logged in, log them out and redirect back to this page
     if auth.logged_in:
         return auth_logout(redirect_url=request.url)
 
-    # Check if request bears a valid verification_key
-    user_obj = get_user(verification_key=verification_key)
-    if not user_obj:
+    # Check if request bears a valid pair of `uid` and `token`
+    user_obj = User.load(uid)
+    if not (user_obj and user_obj.verify_password_token(token=token)):
         error_data = {
-            'message_short': 'Invalid url.',
-            'message_long': 'The verification key in the URL is invalid or has expired.'
+            'message_short': 'Invalid Request.',
+            'message_long': 'The requested URL is invalid, has expired, or was already used',
```

Comment:
update: a better message from product @saradbowman. The message is more informative without leaking the exact failure.